### PR TITLE
did not receive seccomp fd from ring2

### LINKED ISSUE #7696: did not receive seccomp fd from ring2
### Bug description

When starting up new workspace in staging environment, workspace failed with error:
`did not receive seccomp fd from ring2` with `error: "invalid argument"`

It appears that function `receiveSeccmpFd` lacks context when bubbling up error, so not clear which exact return path returned that error.

### Steps to reproduce

Start a new workspace in staging environment

### Workspace affected

_No response_

### Expected behavior

Workspace should start up without errors

### Example repository

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sagor999**
Fixes #7696

```release-note
NONE
```

Edit: [context for the suggested change](https://gitpod.slack.com/archives/C02F19UUW6S/p1642618609068400?thread_ts=1642615433.067400&cid=C02F19UUW6S)

### CODE REVIEW

**Comment by roboquat** [comment]
@sagor999: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **princerachit** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @princerachit` in a comment when ready.

Associated issue: *#7696*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/workspacekit/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/workspacekit/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/workspacekit/cmd/rings.go [hunk 1]
```diff
@@ -669,32 +669,32 @@ func receiveSeccmpFd(conn *net.UnixConn) (libseccomp.ScmpFd, error) {
...
 	_, _, _, _, err = unix.Recvmsg(connfd, nil, buf, 0)
 	if err != nil {
-		return 0, err
+		return 0, xerrors.Errorf("cannot recvmsg from fd '%d': %v", connfd, err)
 	}
 
 	msgs, err := unix.ParseSocketControlMessage(buf)
 	if err != nil {
-		return 0, err
+		return 0, xerrors.Errorf("cannot parse socket control message: %v", err)
```
Comment (princerachit) [inline components/workspacekit/cmd/rings.go:689]: does it make sense to add more info here and other messages in the later if blocks? e.g. `connfd`

**Comment by princerachit** [comment]
@sagor999 it would be helpful if you can edit the title of the PR to reflect the changes that you made.

File: components/workspacekit/cmd/rings.go (hunk 1 above)
Comment (csweichel) [reply components/workspacekit/cmd/rings.go:689]: I don't think so - the message is essentially a binary blob

**Comment by csweichel** [comment]
> @sagor999 it would be helpful if you can edit the title of the PR to reflect the changes that you made.

Agreed :)

nonetheless, the change /lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: a6d27318c6db1a385a27c7d8dac92dd69ec291bc</details>

**Comment by sagor999** [comment]
Fixed the title!
I updated commit message yesterday but forgot to do the same for PR title.

**Pull request merged**